### ISSUE TITLE
[css-tables] Distribution of block size to table sections not interoperable.

### ISSUE THREAD

**Issue by emilio** [w3c/csswg-drafts#10590]
Consider [this test-case](https://bugzilla.mozilla.org/attachment.cgi?id=9413369) from [Mozilla bug 1904559](https://bugzil.la/1904559).

Firefox distributes the size between `<thead>` and `<tbody>` evenly (as if there were two `<tbody>` elements).

Chrome and WebKit prefers growing the `<tbody>` over `<thead>`.

I don't see anything in https://drafts.csswg.org/css-tables/ that implies that `table-header-group` should layout any differently from `table-row-group` (other than the order), but Chrome's behavior seems intentional per https://bugzilla.mozilla.org/show_bug.cgi?id=1904559#c6.

The Chromium / WebKit behavior seems a bit preferable (though it'd be great to see the author feedback @bfgeek mentioned in the comment linked above). I'd be ok with putting it on the spec, but I want to confirm that:

 * Others think the same.
 * @FremyCompany didn't intentionally omit such thing?
 * Would also be good to confirm EdgeHTML behavior for context, it'd be a stronger data point if Edge also did such thing.

**Comment by bfgeek** [comment]
This was one of the issues - https://issues.chromium.org/issues/40770346

But basically its more complex than what that bug suggests.

There are cases (I don't think we could really determine the logic) which FF will prefer a row-group over a header-group/footer-group.

E.g.
```
<!DOCTYPE html>
<table border=1 style="height: 100px;">
  <thead><th style="height: 0px;">header</th></thead>
  <tbody><td>body</td></tbody>
</table>
```
https://www.software.hixie.ch/utilities/js/live-dom-viewer/?saved=12917

Basically for distributing the excess table block-size somehow depends on the contents within it. The algorithm we worked out basically just looks at the height on the {header,footer,row}-groups, and re-distributes based on that vs. determining how to distribute based on the contents. It prefers the row groups over the header/footer groups.

The algorithm is (in simple terms).
1. Grow percent groups up to what their percent specifies (there still may be remaining space after this).
2. With the remaining space grow the following in this preference order.
  1. auto row groups.
  2. fixed row groups.
  3. percent row groups.
  4. auto groups.
  5. fixed groups.
  6. percent groups.

(see also: https://github.com/w3c/csswg-drafts/issues/4418 )

**Comment by kbabbitt** [comment]
> Would also be good to confirm EdgeHTML behavior for context, it'd be a stronger data point if Edge also did such thing.

I don't have a box with pre-Chromium Edge at hand, but I ran the test page in Chromium Edge's IE Mode and found that it matches Firefox's behavior.

Having said that, I agree that the Blink / Webkit behavior seems preferable.

**Comment by FremyCompany** [comment]
Is there an agreement to edit this modified algorithm in the spec?
I'd be happy to.

**Comment by emilio** [comment]
Yeah, I think there is. If we need a resolution we can put it on the agenda for that.
